scripts/ci-entrypoint.sh

```diff
@@ -79,28 +79,22 @@ select_cluster_template() {
         # shellcheck source=scripts/ci-build-kubernetes.sh
         source "${REPO_ROOT}/scripts/ci-build-kubernetes.sh"
         export CLUSTER_TEMPLATE="test/dev/cluster-template-custom-builds.yaml"
-    elif [[ -n "${CI_VERSION:-}" ]] || [[ -n "${USE_CI_ARTIFACTS:-}" ]] || [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+    elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]] || [[ -n "${CI_VERSION:-}" ]]; then
```

Comment:
Not introduced by this PR but could you please fix it? I think I found an issue here.

If `CLUSTER_TEMPLATE` is set, the ci-entrypoint.sh logic leads to `KUBERNETES_VERSION` error like "vlatest".
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/cloud-provider-azure-master-vmss-capz/1529007312948170752

I think the below logic should be moved out of `select_cluster_template()`
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e108534f21de41501a2ec509e9c359ad63bbed7c/scripts/ci-entrypoint.sh#L85-L91
I think `select_cluster_template()` should only choose template with nothing else.

Please correct me if I'm wrong.

Reply:
Thank you Cecile and Jack. Now the request of cloud-provider-azure is to test with "latest-x" and I can update templates in its repo to follow the style in CAPZ.
But I think my suggested change is still needed to make it work? The change won't introduce problems for existing CAPZ templates, right? Meanwhile, it will make cloud-provider-azure's templates work.
Please correct me if I'm wrong.

Reply:
+1 to move k8s version resolution into its own func, independent of select_cluster_template()

Reply:
The current implementation of `ci-entrypoint.sh` is sort of set up to do the following:

1) you may declare a version of kubernetes (including using a "latest" pattern) and we will choose the right template for you

Are we saying we want to introduce the following in addition:

2) you may declare a cluster template and we will choose the right kubernetes version for you

?

Reply:
I think it should support that a user specifies 1) a cluster template and 2) a k8s version (including "latest" pattern), then the script 1) use the template and 2) build k8s if version is of "latest" pattern.

Reply:
"latest" doesn't build kubernetes, it simply pulls the most recent CI bits built and hosted upstream.

If we want any cluster template to support any arbitrary build of Kubernetes then we would have to incorporate all of the foo in this template into every template:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/cluster-template-prow-ci-version.yaml

We don't have a good plan for how to do that at this time.

Reply:
I think most recent built is ok.

Reply:
Thanks for your feedback, I do think this is a problem we want to solve eventually.